translator/trace/jaegerthrift_to_protospan.go

```diff
@@ -0,0 +1,240 @@
+		span := &tracepb.Span{
+			TraceId: jTraceIDToOCProtoTraceID(jspan.TraceIdHigh, jspan.TraceIdLow),
+			SpanId:  jSpanIDToOCProtoSpanID(jspan.SpanId),
+			// TODO: Tracestate: Check RFC status and if is applicable,
+			ParentSpanId: jSpanIDToOCProtoSpanID(jspan.ParentSpanId),
+			Name:         &tracepb.TruncatableString{Value: jspan.OperationName},
+			Kind:         sKind,
+			StartTime:    internal.TimeToTimestamp(startTime),
+			EndTime:      internal.TimeToTimestamp(startTime.Add(time.Duration(jspan.Duration) * time.Microsecond)),
+			Attributes:   sAttributes,
+			// TODO: StackTrace: OpenTracing defines a semantic key for "stack", should we attempt to its content to StackTrace?
+			TimeEvents: jLogsToOCProtoTimeEvents(jspan.Logs),
+			Links:      jReferencesToOCProtoLinks(jspan.References),
+		}
+
+		spans = append(spans, span)
+	}
+
+	return spans
+}
+
+func jLogsToOCProtoTimeEvents(logs []*jaeger.Log) *tracepb.Span_TimeEvents {
+	if logs == nil {
+		return nil
+	}
+
+	timeEvents := &tracepb.Span_TimeEvents{
+		TimeEvent: make([]*tracepb.Span_TimeEvent, 0, len(logs)),
+	}
+	for _, log := range logs {
+		_, attribs := jtagsToAttributes(log.Fields)
+		annotation := &tracepb.Span_TimeEvent_Annotation{
+			Attributes: attribs,
+		}
+		timeEvent := &tracepb.Span_TimeEvent{
+			Time:  internal.TimeToTimestamp(epochMicrosecondsAsTime(uint64(log.Timestamp))),
+			Value: &tracepb.Span_TimeEvent_Annotation_{Annotation: annotation},
+		}
+
+		timeEvents.TimeEvent = append(timeEvents.TimeEvent, timeEvent)
+	}
+
+	return timeEvents
+}
+
+func jReferencesToOCProtoLinks(jrefs []*jaeger.SpanRef) *tracepb.Span_Links {
+	if jrefs == nil {
+		return nil
+	}
+
+	links := &tracepb.Span_Links{
+		Link: make([]*tracepb.Span_Link, 0, len(jrefs)),
+	}
+
+	for _, jref := range jrefs {
+		var linkType tracepb.Span_Link_Type
+		if jref.RefType == jaeger.SpanRefType_CHILD_OF {
+			linkType = tracepb.Span_Link_CHILD_LINKED_SPAN
+		} else {
+			// SpanRefType_FOLLOWS_FROM doesn't map well to OC, so treat all other cases as unknown
+			linkType = tracepb.Span_Link_TYPE_UNSPECIFIED
+		}
+
+		link := &tracepb.Span_Link{
+			TraceId: jTraceIDToOCProtoTraceID(jref.TraceIdHigh, jref.TraceIdLow),
+			SpanId:  jSpanIDToOCProtoSpanID(jref.SpanId),
+			Type:    linkType,
+		}
+		links.Link = append(links.Link, link)
+	}
+
+	return links
+}
+
+func jTraceIDToOCProtoTraceID(high, low int64) []byte {
+	traceID := make([]byte, 16)
+	binary.BigEndian.PutUint64(traceID[:8], uint64(high))
+	binary.BigEndian.PutUint64(traceID[8:], uint64(low))
+	return traceID
+}
+
+func jSpanIDToOCProtoSpanID(id int64) []byte {
+	spanID := make([]byte, 8)
+	binary.BigEndian.PutUint64(spanID, uint64(id))
+	return spanID
+}
+
+func jtagsToAttributes(tags []*jaeger.Tag) (tracepb.Span_SpanKind, *tracepb.Span_Attributes) {
+	if tags == nil {
+		return tracepb.Span_SPAN_KIND_UNSPECIFIED, nil
+	}
+
+	var sKind tracepb.Span_SpanKind
+	sAttribs := &tracepb.Span_Attributes{
```

Comment:
Ditto, please make this `attrsMap` and at the end before return `&tracepb.Span_Attributes{AttributeMap: attrsMap}`